zk/watcher.go

```diff
@@ -310,7 +322,7 @@ func (w *Watcher) createEphemeral(node string) error {
 		}
 	}
 
-	return nil
+	return fmt.Errorf("failed to create ephemeral znode %s after %d retries", node, maxCreateRetries)
```

Comment:
Could we `err` from the final attempt?

Reply:
The final call is `err = w.createAll(parent)`. Only when err == nil, we will reach here.